lab-anthony/lib/basic-auth-parser.js

```diff
@@ -0,0 +1,28 @@
+'use strict';
+
+const httpErrors = require('http-errors');
+const debug = require('debug')('coffee:basic-auth-parser');
+
+module.exports = function(req, res, next){
+  debug('parseBasicAuth');
+  if (!req.headers.authorization){
+    return next(httpErrors(401, 'requires authoriation header'));
+  }
+
+  var authHeader = req.headers.authorization;
+  var namePassword = authHeader.split(' ')[1];
+  namePassword = new Buffer(namePassword, 'base64').toString('utf8');
+  namePassword = namePassword.split(':');
+  req.auth = {
+    username: namePassword[0],
+    password: namePassword[1]
+  };
+
+  if (!req.auth.username){
```

Comment:
It seems weird to me to do error handling at the end, but I don't know how else you would do it.